values/encoding/string_encode.go

```diff
@@ -112,12 +111,25 @@ func (enc *stringEncoder) Encode(strVals values.StringValues, writer io.Writer)
 
 	switch metaProto.Encoding {
 	case encodingpb.EncodingType_RAW_SIZE:
-		return enc.rawSizeEncode(valuesIt, writer)
+		if err := enc.rawSizeEncode(valuesIt, writer); err != nil {
```

Comment:
Why not just `return enc.rawSizeEncode(valuesIt, writer)`?

Reply:
Same note as above, so we can surface the error from calling `compressWriter.Close`.